staging/src/k8s.io/apiserver/pkg/util/flowcontrol/filter-config.go

```diff
@@ -0,0 +1,349 @@
+	if ok := cache.WaitForCacheSync(stopCh, reqMgr.plInformerSynced, reqMgr.fsInformerSynced); !ok {
+		return fmt.Errorf("Never achieved initial sync")
+	}
+	klog.Info("Running reqmgmt config worker")
+	wait.Until(reqMgr.runWorker, time.Second, stopCh)
+	klog.Info("Shutting down reqmgmt config worker")
+	return nil
+}
+
+func (reqMgr *requestManager) runWorker() {
+	for reqMgr.processNextWorkItem() {
+	}
+}
+
+func (reqMgr *requestManager) processNextWorkItem() bool {
+	obj, shutdown := reqMgr.configQueue.Get()
+	if shutdown {
+		return false
+	}
+
+	func(obj interface{}) {
+		defer reqMgr.configQueue.Done(obj)
+		if !reqMgr.syncOne() {
+			reqMgr.configQueue.AddRateLimited(obj)
+		} else {
+			reqMgr.configQueue.Forget(obj)
+		}
+	}(obj)
+
+	return true
+}
+
+// syncOne attempts to sync all the config for the reqmgmt filter.  It
+// either succeeds and returns `true` or logs an error and returns
+// `false`.
+func (reqMgr *requestManager) syncOne() bool {
+	all := labels.Everything()
+	newPLs, err := reqMgr.plLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list PriorityLevelConfiguration objects: %s", err.Error())
+		return false
+	}
+	newFSs, err := reqMgr.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgr.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgr *requestManager) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
+	oldRMState := reqMgr.curState.Load().(*requestManagerState)
+	var shareSum float64
+	newRMState := &requestManagerState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	newlyQuiescent := make([]*priorityLevelState, 0)
+	var haveExemptPL, haveCatchAllPL, haveExemptFS, haveCatchAllFS bool
+	for _, pl := range newPLs {
+		state := oldRMState.priorityLevelStates[pl.Name]
+		if state == nil {
+			state = &priorityLevelState{
+				config: pl.Spec,
+			}
+		} else {
+			oState := *state
+			state = &oState
+			state.config = pl.Spec
+			if state.emptyHandler != nil { // it was undesired, but no longer
+				klog.V(3).Infof("Priority level %q was undesired and has become desired again", pl.Name)
+				state.emptyHandler = nil
+				state.queues.Quiesce(nil)
+			}
+		}
+		if state.config.Limited != nil {
+			shareSum += float64(state.config.Limited.AssuredConcurrencyShares)
+			qsConfig, err := qscOfPL(pl, reqMgr.requestWaitLimit)
+			if err != nil {
+				klog.Warningf(err.Error())
+				continue
+			}
+			state.qsConfig = qsConfig
+		} else {
+			haveExemptPL = true
+		}
+		haveCatchAllPL = haveCatchAllPL || pl.Name == rmtypesv1a1.PriorityLevelConfigurationNameCatchAll
+		newRMState.priorityLevelStates[pl.Name] = state
+	}
+
+	fsSeq := make(apihelpers.FlowSchemaSequence, 0, len(newFSs))
+	for i, fs := range newFSs {
+		_, flowSchemaExists := newRMState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name]
```

Comment:
s/flowSchemaExists/plExists/ ?

Reply:
"flow schema exists in the mind of the beholder"
When the priority level reference is invalid, this code acts as if the flow schema does not exist.

Reply:
renamed